partiql-eval/src/main/java/org/partiql/eval/value/BagValue.java

```diff
@@ -0,0 +1,35 @@
+package org.partiql.eval.value;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.value.PartiQLValueType;
+
+import java.util.Iterator;
+
+/**
+ * This shall always be package-private (internal).
+ */
+class BagValue implements PQLValue {
```

Comment:
I don't think we need Bag/List/Sexp classes if we have a type enum. Because this is always internal, we can add additional classes later, but it doesn't provide additional value. Same for Decimal/DecimalArbitrary.